config/config.go

```diff
@@ -0,0 +1,110 @@
+package config
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/knq/ini"
+	"github.com/knq/ini/parser"
+)
+
+type KV struct {
+	Key   string
+	Value string
+}
+
+type Generator struct {
+	Command string
+	Params  []KV
+}
+
+type Config struct {
+	Dir        string
+	Patterns   []string
+	Generators []Generator
+}
+
+func Load(path string) (*Config, error) {
```

Comment:
I'd rename `path` to `wd` or `startdir` for clarity, especially because I initially read it as "config path"